pytorch_lightning/core/memory.py

```diff
@@ -15,367 +15,7 @@
 import os
 import shutil
 import subprocess
-from collections import OrderedDict
-from typing import Any, Dict, List, Optional, Tuple, Union
-
-import numpy as np
-import torch
-import torch.nn as nn
-from torch import Tensor
-from torch.utils.hooks import RemovableHandle
-
-from pytorch_lightning.utilities import AMPType, DeviceType
-from pytorch_lightning.utilities.imports import _TORCH_GREATER_EQUAL_1_8
-from pytorch_lightning.utilities.warnings import WarningCache
-
-warning_cache = WarningCache()
-
-PARAMETER_NUM_UNITS = [" ", "K", "M", "B", "T"]
-UNKNOWN_SIZE = "?"
-
-
-class LayerSummary(object):
```

Comment:
in case users were importing these directly from pytorch_lightning.core.memory, this will now break. you can add an import like this along with a warning to indicate that imports should be moved to the new utils file: https://github.com/PyTorchLightning/pytorch-lightning/pull/8474/files

Reply:
+1 :)

Reply:
In the current PR state, this will get printed on every lightning import

```python
$ python -c "import pytorch_lightning"
/Users/carlosmocholi/git/pytorch-lightning/pytorch_lightning/core/memory.py:22: LightningDeprecationWarning: pytorch_lightning.core.memory.LayerSummary and pytorch_lightning.core.memory.ModelSummary have been moved to apytorch_lightning.utilities.model_summary since v1.5 and will be removed in v1.7.
```

Reply:
@carmocca do you know how #8474 avoided this?

Reply:
Yes, all the imports for the deprecated mixin were removed first in https://github.com/PyTorchLightning/pytorch-lightning/pull/8396.

Reply:
The complication is that this PR doesn't move the entire file -- just some of the functionality within the file. I specifically left two functions in `core/memory.py` because it didn't make sense to move them, so we would need to find some other solution than what I currently have.

Reply:
@samlurye @carmocca @awaelchli what do you think about moving these other functions from `core/memory.py` to `utilities/memory.py` https://github.com/PyTorchLightning/pytorch-lightning/blob/b7dbcc3e13a2e34e9fc1850d461909caf8ab11e0/pytorch_lightning/core/memory.py#L381-L429

utilities fits these much better. that could be a separate PR to avoid the import noise from the framework